st2actions/tests/unit/test_runner_container.py

```diff
@@ -103,6 +111,23 @@ def test_dispatch(self):
 
         self.assertDictEqual(liveaction_db.context, context)
 
+    @mock.patch.object(LocalShellRunner, 'run', mock.MagicMock(
+        return_value=(action_constants.LIVEACTION_STATUS_SUCCEEDED, NON_UTF8_RESULT, None)))
+    def test_dispatch_non_utf8_result(self):
+        runner_container = get_runner_container()
+        params = {
+            'cmd': "python -c 'print \"\\x82\"'"
+        }
+        liveaction_db = self._get_liveaction_model(RunnerContainerTest.local_action_db, params)
+        liveaction_db = LiveAction.add_or_update(liveaction_db)
+        executions.create_execution_object(liveaction_db)
+
+        try:
+            runner_container.dispatch(liveaction_db)
+            self.fail('Mongo won\'t handle non UTF-8 strings. Should have failed.')
+        except InvalidStringData:
```

Comment:
Minor thing - next time you can just use `self.assertRaises` :P

(yeah, I know this is a late comment)